tests/gradients/test_gradient_transform.py

```diff
@@ -347,6 +347,28 @@ def circuit(weights):
         expected = qml.jacobian(circuit)(weights)
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    def test_setting_shots(self):
+        """Test that setting the number of shots works correctly for
+        a gradient transform"""
+
+        dev = qml.device("default.qubit", wires=1, shots=1000)
+
+        @qml.beta.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        x = 0.543
+
+        # the gradient function can be called with different shot values
+        grad_fn = qml.gradients.param_shift(circuit)
+        assert grad_fn(x).shape == (1, 1)
+        assert grad_fn(x, shots=[(1, 1000)]).shape == (1000, 1)
+
+        # the original QNode is unaffected
+        assert circuit(x).shape == tuple()
+        assert circuit(x, shots=1000).shape == tuple()

```

Comment:
What happens if a circuit uses the shots argument? This was always a tricky case, but shall we define intended behaviour with a test?

Reply:
We currently raise a deprecation warning in the QNode, let me check if it also occurs here, or if one should be added manually 🤔 

https://github.com/PennyLaneAI/pennylane/blob/8a14365e206771874dcfb48ef6d68d51de9118ba/pennylane/qnode.py#L182